packages/itwin/property-grid/src/components/Header.scss

```diff
@@ -4,12 +4,55 @@
 *--------------------------------------------------------------------------------------------*/
 
 .property-grid-react-panel-header {
-  display: flex;
-  align-items: center;
-  box-sizing: border-box;
-  height: calc(var(--iui-size-l) * 2);
+  display: grid;
+  grid-template-areas: "title tools";;
+  grid-template-columns: auto auto;
+  grid-template-rows: auto;
 
   .property-grid-react-header-back-button {
     margin-right: var(--iui-size-xs);
   }
+
+    .header-title {
+      display: flex;
+      justify-content: flex-start;
+      grid-area: title;
+      padding-block: 5px;
+      align-items: center;
+    }
+
+    .header-tools {
+      grid-area: tools;
+      display: flex;
+      justify-content: flex-end;
+      align-items: center;
+    }
+
+    .expandable-search-bar {
+      &.contracted {
+        outline: none;
+
+        &::before {
+          border: none;
+        }
+      }}
+
+    &.search-bar-expanded {
+      display: grid;
+      grid-template-columns: auto;
+      grid-template-rows: auto auto;
+      grid-template-areas:
+      "tools"
+      "title";
+
+      .header-title {
+        grid-area: title
+      }
+
+      .header-tools {
+        grid-area: tools;
+        display: flex;
+        align-items: center;
+      }
```

Comment:
Can be removed. These do not change any styles that are applied when searchbar is collapsed.